src/foam/nanos/auth/Address.js

```diff
@@ -730,21 +732,16 @@ foam.CLASS({
       code: function() {
         return [this.getShortAddress(), this.city, this.regionId, this.countryId, this.postalCode]
           .filter(s => s)
-          .join(', ')
+          .join(', ');
       },
       javaCode: `
-        StringBuilder sb = new StringBuilder();
-        sb.append(getShortAddress());
-        sb.append(", ");
-        sb.append(this.getCity());
-        sb.append(", ");
-        sb.append(getRegionId());
-        sb.append(", ");
-        sb.append(getCountryId());
-        sb.append(", ");
-        sb.append(getPostalCode());
-        String rtn = sb.toString();
-        return rtn.equals(", , , , ") ? "" : rtn;
+        List<String> list = Arrays.asList(getShortAddress(), getCity(), getRegionId(), getCountryId(), getPostalCode()); 
```

Comment:
This is going to be much slower than the original and generate much more garbage. Not an issue on the client since we're using the user's CPU and memory, not ours, and because the client will be doing this on far fewer objects.